src/scripts/libraryBrowser.js

```diff
@@ -1,199 +1,212 @@
+    var dispatchEvent = true;
+
+    if (!views) {
+        dispatchEvent = false;
+        views = button.getAttribute('data-layouts');
+        views = views ? views.split(',') : ['List', 'Poster', 'PosterCard', 'Thumb', 'ThumbCard'];
+    }
+
+    var menuItems = views.map(function (v) {
+        return {
+            name: globalize.translate('Option' + v),
+            id: v,
+            selected: currentLayout == v
+        };
+    });
+
+    import('actionsheet').then(({default: actionsheet}) => {
+        actionsheet.show({
+            items: menuItems,
+            positionTo: button,
+            callback: function (id) {
+                button.dispatchEvent(new CustomEvent('layoutchange', {
+                    detail: {
+                        viewStyle: id
+                    },
+                    bubbles: true,
+                    cancelable: false
+                }));
+
+                if (!dispatchEvent) {
+                    if (window.$) {
+                        $(button).trigger('layoutchange', [id]);
+                    }
+                }
             }
+        });
+    });
+}
+
+export function getQueryPagingHtml (options) {
+    var startIndex = options.startIndex;
+    var limit = options.limit;
+    var totalRecordCount = options.totalRecordCount;
+    var html = '';
+    var recordsEnd = Math.min(startIndex + limit, totalRecordCount);
+    var showControls = limit < totalRecordCount;
+
+    if (html += '<div class="listPaging">', showControls) {
+        html += '<span style="vertical-align:middle;">';
+        html += globalize.translate('ListPaging', (totalRecordCount ? startIndex + 1 : 0), recordsEnd, totalRecordCount);
+        html += '</span>';
+    }
+
+    if (showControls || options.viewButton || options.filterButton || options.sortButton || options.addLayoutButton) {
+        html += '<div style="display:inline-block;">';
+
+        if (showControls) {
+            html += '<button is="paper-icon-button-light" class="btnPreviousPage autoSize" ' + (startIndex ? '' : 'disabled') + '><span class="material-icons arrow_back"></span></button>';
+            html += '<button is="paper-icon-button-light" class="btnNextPage autoSize" ' + (startIndex + limit >= totalRecordCount ? 'disabled' : '') + '><span class="material-icons arrow_forward"></span></button>';
+        }
 
-            return query;
-        },
-        saveQueryValues: function (key, query) {
-            var values = {};
-
-            if (query.SortBy) {
-                values.SortBy = query.SortBy;
-            }
+        if (options.addLayoutButton) {
+            html += '<button is="paper-icon-button-light" title="' + globalize.translate('ButtonSelectView') + '" class="btnChangeLayout autoSize" data-layouts="' + (options.layouts || '') + '" onclick="LibraryBrowser.showLayoutMenu(this, \'' + (options.currentLayout || '') + '\');"><span class="material-icons view_comfy"></span></button>';
```

Comment:
What about `LibraryBrowser.showLayoutMenu`?
I think it is better to extract onclick as listener.

Reply:
Looks like it is impossible - this function just returns HTML.